utilities/transactions/pessimistic_transaction.cc

```diff
@@ -630,7 +630,12 @@ Status PessimisticTransaction::TryLock(ColumnFamilyHandle* column_family,
     // the key is already locked, this func will update some stats on the
     // tracked key. It could also update the tracked_at_seq if it is lower than
     // the existing trackey seq.
-    TrackKey(cfh_id, key_str, tracked_at_seq, read_only, exclusive);
+    //
```

Comment:
The comment above seems to have explained why we still call it even if it is locked?

Reply:
Yeah I explain in the description why I think this is safe (even though we update some stats), but I can elaborate in this comment too.

I guess we also also update tracked_at_seq if the seqno is "earlier" than the tracked ones, but I can't think of a situation where that would happen. I don't think transactions can "go back" in time.

Reply:
I checked the code and it seems fine to me to update the definition of assumed_tracked to be assumed_tracked[_since_last_savepoint]. If it not difficult would be great to throw some asserts to verify that it is indeed tracked properly.